### PR TITLE
feat(webapp): expenses across election rounds

### LINKED ISSUE #25: Expenses for all eden election rounds
As a user I want to see all the expenses that are generated during all the rounds of elections.
create a table and graph that can download the information

### LINKED ISSUE #26: Expenses across eden election rounds
As a user I want to see the total expenses that have been generated during the elections over time.
in a table and graph

### LINKED ISSUE #27: Total Expenses for all delegates
As a user, I would like to see all the expenses of the delegates in a line graph.
this would help to have a tool to have visibility on how money is managed

### LINKED ISSUE #28: Expenses over time
As a user I want to see the total expenses that have been generated, for each specific delegate.
in a graphic representation

### LINKED ISSUE #148: Graph classified and unclassified

### LINKED ISSUE #165: Invalid number of current election
To register the number for the current election it use correct parameter, dont the column 0 of the image:
imagen

### PR DESCRIPTION
**Pull request by warnerHurtado**
### What does this PR do?

- Resolve #165
- Resolve #25 
- Resolve #26 
- Resolve #28 
- Resolve #148 
- Resolve #27

### Steps to test

1. Run the project and open localhost:3000 in your browser
2. click on income and expense and see the new features

#### CheckList

- [X] Follow proper Markdown format
- [X] The content is adequate
- [X] The content is available in both English and Spanish
- [X] I Ran a spell check

### CODE REVIEW

File: webapp/src/components/StackedChartReport/index.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                t('uncategorizedCat')
+          }: ${formatWithThousandSeparator(
+            data.payload[data.dataKey],
+            2
+          )}`}</div>
+        ))}
+    </div>
+  )
+}
+CustomTooltip.propTypes = {
```
Comment (leisterfrancisco) [inline webapp/src/components/StackedChartReport/index.js:53]: @tetogomez it should have a blank line before, shouldn't it?

**Review by leisterfrancisco — APPROVED** [review]
Please, avoid sending big PRs, try to split it into small code changes in the future PRs

File: webapp/src/components/Select/index.js [hunk 2]
```diff
@@ -23,29 +24,25 @@ const IncomeSelect = ({
...
-                : value}
-            </MenuItem>
-          ))
-        ) : (
-          <MenuItem value=""></MenuItem>
-        )}
+        {values.map(value => (
+          <MenuItem key={value} value={value}>
+            {labelSelect === t('textElectionSelect')
+              ? Number(value) + 1
```
Comment (tetogomez) [inline webapp/src/components/Select/index.js:30]: The `ternary if condition` was created to be use as a single line, is a bad practice to use a ternary nested into another ternary

please remove the ternary nested, you can solve it by creating a function to return the correct value for the item

File: webapp/src/components/StackedChartReport/index.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  const { t } = useTranslation('generalForm')
+
+  return (
+    <div>
+      <strong>{label}</strong>
+      {payload &&
+        payload.map((data, i) => (
+          <div key={`${i}-tooltip`}>{`${
+            data.dataKey === 'EXCHANGE_RATE'
+              ? t('exchangeRate')
```
Comment (tetogomez) [inline webapp/src/components/StackedChartReport/index.js:36]: same here, avoid nested ternary condition

File: webapp/src/components/StackedChartReport/index.js (hunk 1 above)
Comment (tetogomez) [reply webapp/src/components/StackedChartReport/index.js:53]: yep, add a blank space

File: webapp/src/components/StackedChartReport/index.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+}
+CustomTooltip.propTypes = {
+  payload: PropTypes.array,
+  label: PropTypes.any,
+  typeCurrency: PropTypes.any
+}
+
+const RenderLegend = ({ payload, typeCurrency }) => {
+  const { t } = useTranslation('generalForm')
+
```
Comment (tetogomez) [inline webapp/src/components/StackedChartReport/index.js:61]: remove this black space, you should be group the const declaration

File: webapp/src/components/StackedChartReport/index.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+            width={18}
+            height={18}
+            ml={2}
+            mt={0.5}
+            mr={0.5}
+            bgcolor={data.color}
+          />
+
+          {data.dataKey === 'EXCHANGE_RATE'
+            ? t('exchangeRate')
```
Comment (tetogomez) [inline webapp/src/components/StackedChartReport/index.js:78]: avoid nested ternary condition

File: webapp/src/components/StackedChartReport/index.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+const StakedChartReport = ({
+  data,
+  firstCategory,
+  secondCategory,
+  typeCurrency,
+  showEosRate
+}) => {
+  const classes = useStyles()
```
Comment (tetogomez) [inline webapp/src/components/StackedChartReport/index.js:105]: same here, group const declaration

BTW: It's a good example because you have three const declarations but one of them is a function, so, you must order it like that:

first const declaration
```
 const classes = useStyles()
 const [getStakedPng, { ref: stackedRef }] = useCurrentPng()
 ```
 then, a blank space and the function declaration

```
const handleStakedDownload = useCallback(async () => {
    const png = await getStakedPng()

    if (png) {
      FileSaver.saveAs(png, 'staked-chart.png')
    }
  }, [getStakedPng])
  ```

File: webapp/src/components/TableReport/index.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import React, { memo, useState } from 'react'
+import PropTypes from 'prop-types'
+import { useTranslation } from 'react-i18next'
+import { DataGrid, esES, enUS } from '@mui/x-data-grid'
+import { createTheme, ThemeProvider } from '@mui/material/styles'
+
+const TableReport = ({ columns, dataPercent }) => {
+  const [pagePaginationSize, setPagePaginationSize] = useState(5)
```
Comment (tetogomez) [inline webapp/src/components/TableReport/index.js:8]: same here, group const declatarion

File: webapp/src/components/TreasuryBalance/index.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import { formatWithThousandSeparator } from '../../utils/format-with-thousand-separator'
+import { useSharedState } from '../../context/state.context'
+
+import styles from './styles'
+
+const useStyles = makeStyles(styles)
+
+const TrasuryBalance = () => {
+  const classes = useStyles()
+  const { t } = useTranslation()
```
Comment (tetogomez) [inline webapp/src/components/TreasuryBalance/index.js:14]: same here

File: webapp/src/hooks/customHooks/useExpenseReportState.js [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+  GET_EXPENSE_TRANSACTIONS_BY_ACCOUNT_QUERY,
+  GET_TOTAL_EXPENSE_BY_ELECTIONS_QUERY,
+  GET_PERCENT_EXPENSES_ALL_ELECTIONS,
+  GET_PERCENT_EXPENSES_BY_ELECTION,
+  GET_EXPENSE_ELECTIONS_BY_YEAR,
+  GET_TOTAL_EXPENSE_BY_CATEGORY
+} from '../../gql'
+
+const useExpenseReport = () => {
+  const [showElectionRadio, setShowElectionRadio] = useState('allElections')
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useExpenseReportState.js:25]: group const declarations

File: webapp/src/hooks/customHooks/useTresuryBalanceState.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+import { eosApi } from '../../utils/eosapi'
+
+const getActualDate = () => {
+  const date = new Date()
+
+  return `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`
+}
+
+const useTresuryBalanceState = () => {
+  const [currencyBalance, setCurrencyBalance] = useState('')
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useTresuryBalanceState.js:13]: same here

File: webapp/src/routes/ExpenseReport/index.js [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+import SelectComponent from '../../components/Select'
+
+import styles from './styles'
+
+const useStyles = makeStyles(styles)
+
+const rowsCenter = { flex: 1, align: 'center', headerAlign: 'center' }
+
+const ExpenseReport = () => {
+  const classes = useStyles()
```
Comment (tetogomez) [inline webapp/src/routes/ExpenseReport/index.js:30]: group const declaration

File: webapp/src/routes/ExpenseReport/index.js [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+
+  const { t } = useTranslation('expenseRoute')
+
+  const { t: t2 } = useTranslation('generalForm')
```
Comment (tetogomez) [inline webapp/src/routes/ExpenseReport/index.js:34]: why do you need to rename the translate function?

do you know you can use the `useTranslation` hook without a parameter and in this way get access to entire language props?

So, you can do that: `const { t } = useTranslation()` and if you need to access the `expenseRoute` properties just need to use the `t` function with the complete language property path, for example:

`t('expenseRoute. textInformation')`

And the same way for `generalForm `: 

`t('generalForm. textRadioButton3')`

In short, using the t function with a complete path we are avoiding calling twice the useTranslate hook

**Review by tetogomez — CHANGES_REQUESTED** [review]
please check the comments

File: webapp/src/routes/ExpenseReport/index.js (hunk 12 above)
Comment (JAMares) [reply webapp/src/routes/ExpenseReport/index.js:34]: I tried with the complete path and dont work, but i searched and find [this ](https://react.i18next.com/latest/usetranslation-hook) like next image:
imagen
With this i t2 deleted and work fine

Comment (JAMares) [reply webapp/src/routes/ExpenseReport/index.js:34]: I tried this:
`const { t } = useTranslation()`
`headerName: t('tableHeader2', { ns: 'expenseRoute' })`
Where "ns" is the path, and work fine

**Review by tetogomez — APPROVED** [review]
LGTM
